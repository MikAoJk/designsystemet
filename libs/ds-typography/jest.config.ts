export default {
  displayName: 'ds-typography',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-typography',
};
