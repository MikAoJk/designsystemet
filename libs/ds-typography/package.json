{
  "name": "@skatteetaten/ds-typography",
  "version": "0.0.1",
  "groupId": "no.skatteetaten.aurora",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": "^18"
  }
}
