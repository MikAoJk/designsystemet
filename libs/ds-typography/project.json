{
  "name": "ds-typography",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-typography/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-typography",
        "tsConfig": "libs/ds-typography/tsconfig.lib.json",
        "project": "libs/ds-typography/package.json",
        "entryFile": "libs/ds-typography/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-typography/rollup.config.js"],
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/ds-typography/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/ds-typography/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-typography/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-typography/**/*.css",
          "libs/ds-typography/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-typography"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish {args.switches}",
        "cwd": "dist/libs/ds-typography"
      }
    }
  },
  "implicitDependencies": ["!ds-dev-config"]
}
