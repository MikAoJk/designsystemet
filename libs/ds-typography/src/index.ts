export * from './Blockquote/Blockquote';
export * from './Blockquote/Blockquote.types';
export * from './Paragraph/Paragraph';
export * from './Paragraph/Paragraph.types';
