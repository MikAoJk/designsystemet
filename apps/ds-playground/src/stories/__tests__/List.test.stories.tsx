import { AxePuppeteer } from '@axe-core/puppeteer';
import { List, listAsArr, ListProps } from '@skatteetaten/ds-typography';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { toHaveNoViolations } from 'jest-axe';
import { Page } from 'puppeteer';

import {
  screenShotOptions,
  wrapper,
} from './testUtils/puppeteer.testing.utils';

const verifyMatchSnapShot = async (page: Page): Promise<void> => {
  const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
  expect(innerHtml).toMatchSnapshot();
};

const verifyMatchImageSnapShot = async (page: Page): Promise<void> => {
  const image = await page.screenshot(screenShotOptions);
  expect(image).toMatchImageSnapshot();
};

const verifyAxeRules = async (page: Page): Promise<void> => {
  const axeResults = await new AxePuppeteer(page).include(wrapper).analyze();
  expect.extend(toHaveNoViolations);
  expect(axeResults).toHaveNoViolations();
};

const verifySnapshotsAndAxeRules = async (page: Page): Promise<void> => {
  await verifyMatchSnapShot(page);
  await verifyMatchImageSnapShot(page);
  await verifyAxeRules(page);
};

export default {
  component: List,
  title: 'Tests / List',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
    },
    hasIndent: { table: { disable: true } },
    hasSpacing: { table: { disable: true } },
    // Tags
    as: {
      table: { disable: true },
      options: [...listAsArr],
      control: 'inline-radio',
    },
  },
} as ComponentMeta<typeof List>;

const defaultArgs: ListProps = {
  children: [
    <List.Element key={'listElement_1'}>
      {'Kjenner du behovet til brukeren?'}
    </List.Element>,
    <List.Element key={'listElement_2'}>
      {'Er du sikker på at du kjenner behovet til brukeren?'}
    </List.Element>,
    <List.Element key={'listElement_3'}>{'Snakk med andre.'}</List.Element>,
  ],
};
const Template: ComponentStory<typeof List> = (args) => (
  <div data-test-block>
    <p>
      {'Jeg er et tekstavsnitt uten innrykk. Denne teksten skulle bli ca. to linjer lang, men jeg trenger litt mer innhold for å bli to linjer lang. ' +
        'Derfor fortsetter jeg å skrive på dette tekstavsnittet uten innrykk og etter en stund blir denne teksten ca to linjer lang.'}
    </p>
    <List {...args} />
    <p>
      {'Jeg er et tekstavsnitt uten innrykk. Denne teksten skulle bli ca. to linjer lang, men jeg trenger litt mer innhold for å bli to linjer lang. ' +
        'Derfor fortsetter jeg å skrive på dette tekstavsnittet uten innrykk og etter en stund blir denne teksten ca to linjer lang.'}
    </p>
  </div>
);

// Når List har en ref, så får dom elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.argTypes = {
  ...WithRef.argTypes,
  ref: { table: { disable: false } },
};
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLUListElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const refId = await page.$eval(`${wrapper} > ul`, (el) => el.id);
    expect(refId).toBe('dummyIdForwardedFromRef');
  },
};

// Når List har en id, så har elementet id'en satt
export const WithId = Template.bind({});
WithId.storyName = 'With Id (FA2)';
WithId.argTypes = {
  ...WithId.argTypes,
  id: { table: { disable: false } },
};
WithId.args = {
  ...defaultArgs,
  id: 'ListId',
};
WithId.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const id = await page.$eval(`${wrapper} > ul`, (el) =>
      el.getAttribute('id')
    );
    expect(id).toBe('ListId');
  },
};

// Når List har en custom CSS, så vises custom stil
export const WithCustomCss = Template.bind({});
WithCustomCss.storyName = 'With Custom CSS (FA3)';
WithCustomCss.args = {
  ...defaultArgs,
  className: 'dummyClassname',
};
WithCustomCss.argTypes = {
  ...WithCustomCss.argTypes,
  className: { table: { disable: false } },
};
WithCustomCss.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifySnapshotsAndAxeRules(page);

    const classNameAttribute = await page.$eval(`${wrapper} > ul`, (el) =>
      el.getAttribute('class')
    );
    expect(classNameAttribute).toContain('dummyClassname');
  },
};

// Når List har en lang, så har elementet lang satt
export const WithLang = Template.bind({});
WithLang.storyName = 'With Lang (FA4)';
WithLang.argTypes = {
  ...WithLang.argTypes,
  lang: { table: { disable: false } },
};
WithLang.args = {
  ...defaultArgs,
  lang: 'nb',
};
WithLang.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const langAttribute = await page.$eval(`${wrapper} > ul`, (el) =>
      el.getAttribute('lang')
    );
    expect(langAttribute).toBe('nb');
  },
};

// Når List har dataTestid, så har elementet data-testid satt
export const WithDataTestid = Template.bind({});
WithDataTestid.storyName = 'With DataTestid (FA5)';
WithDataTestid.argTypes = {
  ...WithDataTestid.argTypes,
  'data-testid': { table: { disable: false } },
};
WithDataTestid.args = {
  ...defaultArgs,
  'data-testid': 'ListID',
};
WithDataTestid.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const dataTestid = await page.$eval(`${wrapper} > ul`, (el) =>
      el.getAttribute('data-testid')
    );
    expect(dataTestid).toBe('ListID');
  },
};

// Når List instansieres, får den riktige default-verdier
export const Defaults = Template.bind({});
Defaults.storyName =
  'Defaults Variant Bullet (A1 - 1 av 2, B1 - 1 av 2, B2 - 1 av 2)';
Defaults.argTypes = {
  ...Defaults.argTypes,
  children: {
    table: { disable: false },
    control: { type: null },
  },
};
Defaults.args = {
  ...defaultArgs,
};
Defaults.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifySnapshotsAndAxeRules(page);

    const listElement = await page.$(`${wrapper} > ul`);
    expect(listElement).toBeTruthy();
  },
};

// Når List har en as, får elementet riktig tag og ser riktig ut
export const VariantNumber = Template.bind({});
VariantNumber.storyName = 'Variant Number(A1 - 2 av 2, B1 - 2 av 2)';
VariantNumber.argTypes = {
  ...VariantNumber.argTypes,
  as: {
    table: { disable: false },
  },
};
VariantNumber.args = {
  ...defaultArgs,
  as: 'ol',
};
VariantNumber.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifySnapshotsAndAxeRules(page);

    const listElement = await page.$(`${wrapper} > ol`);
    expect(listElement).toBeTruthy();
  },
};

// Når List instansieres med markup, får markup riktig styling
export const WithMarkup = Template.bind({});
WithMarkup.storyName = 'With Markup (A2, B2 - 2 av 2)';
WithMarkup.argTypes = {
  ...WithMarkup.argTypes,
  children: {
    table: { disable: false },
    control: { type: null },
  },
};
WithMarkup.args = {
  ...defaultArgs,
  children: [
    <List.Element key={'listElement_1'}>
      {'Kjenner du behovet til '}
      <strong>{'brukeren?'}</strong>
    </List.Element>,
    <List.Element key={'listElement_2'}>
      {'Er du sikker på at du kjenner '}
      <a href={'https://skatteetaten.no'}>{'behovet til brukeren?'}</a>
    </List.Element>,
    <List.Element key={'listElement_3'}>
      {'Snakk med andre '}
      <em>{'italic '}</em>
      <mark>{'mark'}</mark>
    </List.Element>,
  ],
};
WithMarkup.parameters = {
  puppeteerTest: verifySnapshotsAndAxeRules,
};

// Når List har spacing, så får elementet en margin under listen
export const WithSpacing = Template.bind({});
WithSpacing.storyName = 'With Spacing (A3)';
WithSpacing.argTypes = {
  ...WithSpacing.argTypes,
  hasSpacing: { table: { disable: false } },
};
WithSpacing.args = {
  ...defaultArgs,
  hasSpacing: true,
};
WithSpacing.parameters = {
  puppeteerTest: verifySnapshotsAndAxeRules,
};

// Når List ikke har innrykk, så får elementet ingen margin foran listelementene
export const WithoutIndent = Template.bind({});
WithoutIndent.storyName = 'Without Indent (A4)';
WithoutIndent.argTypes = {
  ...WithoutIndent.argTypes,
  hasIndent: { table: { disable: false } },
};
WithoutIndent.args = {
  ...defaultArgs,
  hasIndent: false,
};
WithoutIndent.parameters = {
  puppeteerTest: verifySnapshotsAndAxeRules,
};
