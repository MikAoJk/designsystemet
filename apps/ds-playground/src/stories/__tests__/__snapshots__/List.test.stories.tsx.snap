// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/List Defaults smoke-test 1`] = `
<div data-test-block="true">
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA  ">
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
  </ul>
</div>
`;

exports[`Tester/List VariantNumber smoke-test 1`] = `
<div data-test-block="true">
  <ol class="List_list__xzA1c   ">
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
  </ol>
</div>
`;

exports[`Tester/List WithBothVariantSpacingAndAtLeast10NumberItems smoke-test 1`] = `
<div data-test-block="true">
  <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf  ">
    Her kan du se hvordan de forskjellige variantene til en list ser ut sammen med andre elementer med veksling av luft under listene.
  </p>
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA  ">
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet?
    </li>
  </ul>
  <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf  ">
    Den første listen kan veksle mellom variantene en liste har.
  </p>
  <ol class="List_list__xzA1c   ">
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre
    </li>
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre når som helst
    </li>
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre ofte
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre aldri
    </li>
  </ol>
  <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf  ">
    Videre så er en liste med lenker også tatt med for å se hvordan det hele ser ut sammen.
  </p>
  <ul class="LinkGroup_linkGroup__ZiEgs   ">
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="LinkGroup_linkGroupItem__Y7SjF">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
        </path>
      </svg>
      <a class="Link_link__jeUpY  "
         href="#root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/List WithCustomCss smoke-test 1`] = `
<div data-test-block="true">
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA  dummyClassname">
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
  </ul>
</div>
`;

exports[`Tester/List WithDataTestid smoke-test 1`] = `
<div data-test-block="true">
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA  "
      data-testid="ListID"
  >
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
  </ul>
</div>
`;

exports[`Tester/List WithId smoke-test 1`] = `
<div data-test-block="true">
  <ul id="ListId"
      class="List_list__xzA1c List_list_withBullet__s5lqA  "
  >
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
  </ul>
</div>
`;

exports[`Tester/List WithLang smoke-test 1`] = `
<div data-test-block="true">
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA  "
      lang="nb"
  >
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
  </ul>
</div>
`;

exports[`Tester/List WithLongTextAndBreaking smoke-test 1`] = `
<div data-test-block="true">
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA  ">
    <li class="ListElement_element__mEewo ">
      Denne listItem har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert, brekke over flere linjer og får et hengende innrykk.
    </li>
    <li class="ListElement_element__mEewo ">
      DennelistItemharenveldiglangtekst.Sålangatdentvingerframlinjeskiftmedtekstsomalltidervenstrejustertbrekkeoverflerelinjerogfårethengendeinntrykk.
    </li>
  </ul>
</div>
`;

exports[`Tester/List WithMarkup smoke-test 1`] = `
<div data-test-block="true">
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA  ">
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til
      <strong>
        brukeren?
      </strong>
    </li>
    <li class="ListElement_element__mEewo ">
      Behovet til brukeren er veldig viktig.
      <a href="#root">
        Er du sikker på at du kjenner behovet til brukeren?
      </a>
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre om
      <em>
        italic,
      </em>
      <mark>
        mark og
      </mark>
      <code>
        code blokk
      </code>
      bare på gøy
    </li>
  </ul>
</div>
`;

exports[`Tester/List WithRef smoke-test 1`] = `
<div data-test-block="true">
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA  "
      id="dummyIdForwardedFromRef"
  >
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
  </ul>
</div>
`;

exports[`Tester/List WithSpacing smoke-test 1`] = `
<div data-test-block="true">
  <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf Paragraph_paragraph_hasSpacingLarge__6x1nL ">
    Jeg er et tekstavsnitt uten innrykk. Denne teksten skulle bli ca. to linjer lang, men jeg trenger litt mer innhold for å bli to linjer lang. Derfor fortsetter jeg å skrive på dette tekstavsnittet uten innrykk og etter en stund blir denne teksten ca to linjer lang.
  </p>
  <ul class="List_list__xzA1c List_list_withBullet__s5lqA List_list_hasSpacing__a4Kcg ">
    <li class="ListElement_element__mEewo ">
      Kjenner du behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Er du sikker på at du kjenner behovet til brukeren?
    </li>
    <li class="ListElement_element__mEewo ">
      Snakk med andre.
    </li>
  </ul>
  <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf  ">
    Jeg er et tekstavsnitt uten innrykk. Denne teksten skulle bli ca. to linjer lang, men jeg trenger litt mer innhold for å bli to linjer lang. Derfor fortsetter jeg å skrive på dette tekstavsnittet uten innrykk og etter en stund blir denne teksten ca to linjer lang.
  </p>
</div>
`;
